server/src/LspServer.ts

```diff
@@ -149,25 +134,74 @@ export class LspServer {
       // Register for all configuration changes.
       await this.connection.client.register(DidChangeConfigurationNotification.type, undefined);
     }
+
+    const profileResult = this.dbtProfileCreator.createDbtProfile();
+    const contextInfo = profileResult.isOk()
+      ? { type: profileResult.value.type, method: profileResult.value.method }
+      : { type: profileResult.error.type, method: profileResult.error.method };
```

Comment:
I'm not sure but can we simplify it like this: `const contextInfo = profileResult.match<DbtProfileResult>(s => s, e => e);`

Reply:
Refactored